language/move-lang/tests/move_check/parser/attribute_placement.move

```diff
@@ -0,0 +1,46 @@
+#[attr]
+address 0x42 {
+#[attr]
+module M {
+    #[attr]
+    use 0x42::N;
+
+    #[attr]
+    struct S {}
+
+    #[attr]
+    const C: u64 = 0;
+
+    #[attr]
+    public fun foo() { N::bar() }
+
+    #[attr]
+    spec fun foo {}
+}
+}
+
+#[attr]
+module 0x42::N {
+    #[attr]
+    friend 0x42::M;
+
+    #[attr]
+    public fun bar() {}
+}
+
+#[attr]
```

Comment:
Realize this is a draft so probably more tests incoming, but just mentioning here since it would be nice to test:
* Multiple attributes (both same, and different)
* Parametrized attributes

Reply:
See attribute_variants.move, but I'll update that one to handle the same/different case